gensim/scripts/segment_wiki.py

```diff
@@ -5,12 +5,14 @@
 # Copyright (C) 2016 RaRe Technologies
 
 """
-Construct a corpus from a Wikipedia (or other MediaWiki-based) database dump and extract sections of pages from it
-and save to json-line format.
+Construct a corpus from a Wikipedia (or other MediaWiki-based) database dump (typical filename
+is <LANG>wiki-<YYYYMMDD>-pages-articles.xml.bz2 or <LANG>wiki-latest-pages-articles.xml.bz2),
+extract titles, section names, section content and save to json-line format,
+that contains 3 fields ::
 
-If you have the `pattern` package installed, this module will use a fancy
-lemmatization to get a lemma of each token (instead of plain alphabetic
-tokenizer). The package is available at https://github.com/clips/pattern .
+    'title' (str) - title of article,
+    'section_titles' (list) - list of titles of sections,
+    'section_texts' (list) - list of content from sections.
```

Comment:
I'd prefer to include a concrete example, something like this:

---

Process a raw Wikipedia dump (XML.bz2 format, for example https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-pages-articles.xml.bz2 for the English Wikipedia) and extract all articles and their sections as plain text::

    python -m gensim.scripts.segment_wiki -f enwiki-20171001-pages-articles.xml.bz2 -o enwiki-20171001-pages-articles.json.gz

The output format of the parsed Wikipedia is *json-lines* = one article per line, serialized into JSON. Here's an example how to load it in Python::

    # iterate over the file we just created
    for line in smart_open('enwiki-20171001-pages-articles.txt.gz'):
        # decode JSON into a Python object
        article = json.loads(line)

        # each article has a "title", "section_titles" and "section_texts" fields
        print("Article title: %s" % article['title']) 
        for section_title, section_text in zip(article['section_titles'], article['section_texts']):
            print("Section title: %s" % section_title)
            print("Section text: %s" % section_text)